pallets/utxo/src/lib.rs

```diff
@@ -903,11 +1078,46 @@ pub mod pallet {
 
             spend::<T>(&signer, &tx)
         }
+
+        /// unlock the stake, if user wants to stop validating and withdraw the locked utxos.
+        /// If used with `pallet-staking`, it uses the `BondingDuration`
+        /// to set the period/era on when to withdraw.
+        #[pallet::weight(T::WeightInfo::unlock_stake(1 as u32))]
+        pub fn unlock_stake(
+            origin: OriginFor<T>,
+            controller_pubkey: H256
+        ) -> DispatchResultWithPostInfo {
+            ensure_signed(origin)?;
+            staking::unlock::<T>(&controller_pubkey)?;
+            Ok(().into())
+        }
+
+        /// withdraw unlocked stake. Make sure the period for withdrawal has passed.
+        /// If used with `pallet-staking`, then the period can be found in
+        /// its ledger of datatype `StakingLedger`,
+        /// the field `unlocking` of datatype `UnlockChunk`,
+        /// and at field `era`.
+        #[pallet::weight(T::WeightInfo::withdraw_stake(outpoints.len() as u32))]
+        pub fn withdraw_stake(
+            origin: OriginFor<T>,
+            controller_pubkey: H256,
+            outpoints: Vec<H256>
+        ) -> DispatchResultWithPostInfo {
+            ensure_signed(origin)?;
+            staking::withdraw::<T>(controller_pubkey,outpoints)?;
```

Comment:
Same here about the controller overcontrol. If I understand correctly, a controller should only be able to stake. He shouldn't be allowed to spend the money (except for staking, which is a form of spending, depending on the protocol, which I'm not familiar with). So, spending otherwise (or withdrawal here) should only be authorized if the stash key signs the transaction to do this.